### PR TITLE
Add getElementScreen

### PR DESCRIPTION
**Pull request by zcmgyu**
In Java client I could get screenshot of element, but WD wasn't implemented this method, though [getElementScreen API](https://github.com/appium/appium-base-driver/blob/0765c92d7b5c7dc9cf5273fd5a3c4546e4281493/lib/protocol/routes.js#L581)'s also prepared already.
> '/wd/hub/session/:sessionId/element/:elementId/screenshot': {
>     GET: {command: 'getElementScreenshot'}
> }

**Usage:** 
> await element.getElementScreen();

or

> await driver.getElementScreen(element);

### CODE REVIEW

File: lib/commands.js [hunk 1]
```diff
@@ -1100,6 +1100,70 @@ commands.takeScreenshot = function() {
...
+    }]);
+  });
+};
+
+/**
+ * clickElement(element, cb) -> cb(err)
+ *
+ * @jsonWire POST /session/:sessionId/element/:id/click
+ */
+commands.clickElement = function(element) {
```
Comment (dmlemeshko) [inline lib/commands.js]: This method already implemented and exist in your file: https://github.com/zcmgyu/wd/blob/c9cb7519e728bbbf3214165d0a1ceed5f5d0668d/lib/commands.js#L1158

File: lib/element-commands.js [hunk 2]
```diff
@@ -432,4 +432,13 @@ elementCommands.setImmediateValueInApp = function(value, cb) {
  */
 elementCommands.setImmediateValue = elementCommands.setImmediateValueInApp;
 
+/**
+ * element.getElementScreenshot(cb) -> cb(err)
+ *
+ * @jsonWire POST /session/:sessionId/element/:id/getElementScreenshot
+ */
+elementCommands.getElementScreenshot = function (cb) {
```
Comment (jlipps) [inline lib/element-commands.js]: probably just call it `getScreenshot` since you'll be calling it on an element.

File: lib/commands.js [hunk 3]
```diff
@@ -1100,6 +1100,70 @@ commands.takeScreenshot = function() {
...
+    } else {
+      if(path.extname(_path) === '') { _path = _path + '.png'; }
+      cb(null, _path);
+    }
+  }
+
+  buildFilePath(_path, function(err, filePath) {
+    commands.getElementScreenshot.apply(_this, [function(err, base64Data) {
+      if(err) { return cb(err); }
+      require("fs").writeFile(filePath, base64Data, 'base64', function(err) {
```
Comment (jlipps) [inline lib/commands.js]: please require `fs` up top

**Comment by jlipps** [comment]
@zcmgyu thank you for this. I'd prefer to review this without all the extraneous styling changes, however. Is that something you did manually or did it get reformatted by your IDE?

**Comment by admc** [comment]
Greetings, when I run our unit tests "gulp test-midway-chrome", with selenium jar 3.9.1 I get the following error with your branch:

1) element (local, chrome) element.sendKeys:
     AssertionError: expected '/Users/adam/projects/wd/test/mocha.opts' to not include '/Users/adam/projects/wd/test/mocha.opts'
      at test/midway/element-specs.js:225:28
      at _fulfilled (node_modules/q/q.js:834:54)
      at node_modules/q/q.js:863:30
      at Promise.promise.promiseDispatch (node_modules/q/q.js:796:13)
      at node_modules/q/q.js:604:44
      at runSingle (node_modules/q/q.js:137:13)
      at flush (node_modules/q/q.js:125:13)
      at process._tickCallback (internal/process/next_tick.js:61:11)

Thoughts?

**Pull request closed without merging**